libraries/ESP8266WiFi/src/ESP8266WiFiSTA.cpp

```diff
@@ -119,10 +119,12 @@ wl_status_t ESP8266WiFiSTAClass::begin(const char* ssid, const char *passphrase,
     }
 
     struct station_config conf;
-    strcpy(reinterpret_cast<char*>(conf.ssid), ssid);
+    if(strlen(ssid) == 32)

```

Comment:
`strncpy(reinterpret_cast<char*>(conf.ssid), ssid, 32);` will do the same w/o any if required.

From the man page, this is one of the intended uses:
>        One  valid  (and  intended) use of strncpy() is to copy a C string to a fixed-length buffer while ensuring both that the buffer is not overflowed and that unused bytes in the target buffer are zeroed out (perhaps to prevent information leaks if the buffer is to be written to media or transmitted to another process via an interprocess communication technique).